npm/translatePolicy.go

```diff
@@ -936,7 +936,7 @@ func translateEgress(ns string, policyName string, targetSelector metav1.LabelSe
 					if portRuleExists {
 						for _, portRule := range rule.Ports {
 							if portRule.Port != nil && portRule.Port.IntValue() == 0 {
-								portName := portRule.Port.String()
+								portName := util.NamedPortIPSetPrefix + portRule.Port.String()
```

Comment:
Can we confirm this? We can still have a port rule without a named port, right?

Reply:
Okay checked the logic, all port strings are getting appending to Namedports and in the end we call util.DropEmptyFields(namedPorts) (Line: 1438). This is weird. But with the updated logic empty port strings will not be dropped. We should only add if portRule.port.String() is not empty. This should have been the logic at first place instead of double parsing :-/

Reply:
We should still add a test with policy which has port rules (without named ports) and one without

Reply:
NPM is allowing this below network policy, corrected the behavior now.

```
  ingress:
    - from:
        - podSelector:
            matchLabels:
              app: backend
      ports:
        - port: ""
```